src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -91,13 +92,31 @@ class IqPolicyEvaluatorUtil
     if (!isNetworkError || failBuildOnNetworkError) {
       throw e
     }
+    else if (isHttp404Error(e)) {
+      listener.logger.println Messages._IqPolicyEvaluation_ResourceNotFound()
+      run.result = Result.FAILURE
+      return null
+    }
     else {
       listener.logger.println ExceptionUtils.getStackTrace(e)
       run.result = Result.UNSTABLE
       return null
     }
   }
 
+  private static boolean isHttp404Error(final Exception throwable) {
+    boolean result = false
+    int httpResponseExceptionIndex = ExceptionUtils.indexOfType(throwable, HttpResponseException)
+    if (httpResponseExceptionIndex >= 0) {
+      Throwable[] throwables = ExceptionUtils.getThrowables(throwable)
+      HttpResponseException httpResponseException = (HttpResponseException) throwables[httpResponseExceptionIndex]
```

Comment:
I'm being lazy and not looking this up myself on the hope that you know from your research. Do we know for sure that this is going to be a HttpResponseException or do we need to do an instanceof check before casting it?

Reply:
> Do we know for sure that this is going to be a HttpResponseException

Yes, I know for sure.   The call ```int httpResponseExceptionIndex = ExceptionUtils.indexOfType(throwable, HttpResponseException)``` is why i know for sure.  I first check is it a ```HttpResponseException``` by looking if it is a member of the throwables.  If yes i then grab it and cast it.